### PR TITLE
GraphQL: Move *ItemInterface docs to their own topics

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) moves attributes related to the CreditMemoItemInterface, DownloadableItemInterface, OrderItemInterface, and ShipmentItemInterface from the `Customer` output object into their own topics. 

## Affected DevDocs pages

-  ...

## Links to Magento source code

-  ...

### CODE REVIEW

File: src/_includes/graphql/customer-orders-output.md [hunk 1]
```diff
@@ -107,26 +75,17 @@ Attribute | Data type | Description
 --- | --- | ---
 `comments` | [[SalesCommentItem](#SalesCommentItem)] | Comments on the credit memo
```
Comment (rogyar) [inline src/_includes/graphql/customer-orders-output.md:76]: It's not related to the current PR, but just to mention that we have some changes introduced in this schema 

```suggestion
`comments` | [[CommentItem](#SalesCommentItem)] | Comments on the credit memo
```

https://github.com/magento/magento2/blob/20b7e0d503f1b50fe2ca7385cf41657501b47398/app/code/Magento/SalesGraphQl/etc/schema.graphqls#L56

**Comment by keharper** [comment]
Thanks for your review @rogyar. I've made the correction.

**Review by dobooth — APPROVED** [review]

File: src/_includes/graphql/customer-orders-output.md [hunk 2]
```diff
@@ -67,66 +67,25 @@ Attribute | Data type | Description
...
-`bundle_options` | [[ItemSelectedBundleOption]](#ItemSelectedBundleOption) | A list of bundle options that are assigned to the bundle product
-
 #### CreditMemo attributes {#CreditMemo}
 
 The `CreditMemo` object contains details about credit memos applied to an order.
 
 Attribute | Data type | Description
 --- | --- | ---
-`comments` | [[SalesCommentItem](#SalesCommentItem)] | Comments on the credit memo
+`comments` | [[CommentItem](#SalesCommentItem)] | Comments on the credit memo
```
Comment (prabhuram93) [inline src/_includes/graphql/customer-orders-output.md:76]: `[[CommentItem](#SalesCommentItem)]` I don't understand clearly how this is interpreted, but the comments should be of the type SalesCommentItem.

File: src/_includes/graphql/downloadable-items-links.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#### DownloadableItemsLinks attributes {#DownloadableItemsLinks}
+
+The `DownloadableProductLinks` object defines characteristics of a downloadable product.
```
Comment (prabhuram93) [inline src/_includes/graphql/downloadable-items-links.md:3]: I think this should be `DownloadableItemsLinks`. @keharper you can double check.

File: src/_includes/graphql/invoice-item-interface.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Attribute | Data type | Description
+--- | --- | ---
+`discounts` | [Discount] | Contains information about the final discount amount for the base product, including discounts on options
+`id` | ID! | The unique ID of the invoice item
+`order_item` | OrderItemInterface | Contains details about an individual order item
+`product_name` | String | The name of the base product
+`product_sale_price` | Money! | The sale price for the base product including selected options
+`product_sku` | String! | The SKU of the base product
+`product_type` | String | The type of product, such as simple, configurable, or bundle
```
Comment (prabhuram93) [inline src/_includes/graphql/invoice-item-interface.md:9]: @keharper I don't think we are returning `product_type`. Please check.

File: src/_includes/graphql/item-selected-bundle-option.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#### ItemSelectedBundleOption attributes {#ItemSelectedBundleOption}
+
+The ItemSelectedBundleOption object contains a list of bundle options that are assigned to the bundle product.
+
+Attribute | Data type | Description
+--- | --- | ---
+`id` | ID! | The unique identifier of the option
+`label` | String! | The label of the option
+`values` | [[ItemSelectedBundleOptionValue!](#ItemSelectedBundleOptionValue)]! | A list of products that represent the values of the parent option
```
Comment (prabhuram93) [inline src/_includes/graphql/item-selected-bundle-option.md:9]: `ItemSelectedBundleOptionValue!` is not mandatory in the code. I don't see any reason why it should not be mandatory and is probably a bug. But for documenting lets reflect what we are shipping. I will log a P3 for this.

Comment (keharper) [reply src/_includes/graphql/item-selected-bundle-option.md:9]: @prabhuram93 The docs are wrong. The object was probably required at some point in 2.4.1.

File: src/_includes/graphql/customer-orders-output.md (hunk 2 above)
Comment (keharper) [reply src/_includes/graphql/customer-orders-output.md:76]: SalesCommentItem changes to CommentItem in 2.4.2. I'll create a ticket to change those instances.

**Review by prabhuram93 — APPROVED** [review]

**Comment by keharper** [comment]
running tests

**Pull request merged**